questionnaire/dataset/dataset-service.js

```diff
@@ -98,22 +98,81 @@ function createDatasetService({
         progress.forEach(sectionId => {
             const questionAnswers = answers[sectionId];
 
-            if (sectionId === 'p-applicant-declaration' && questionnaire.version === '5.2.1') {
+            if (
+                [
+                    'p-applicant-infections',
+                    'p-applicant-pregnancy',
+                    'p-applicant-pregnancy-loss'
+                ].includes(sectionId) &&
+                questionnaire.version === '5.2.1'
+            ) {
+                // const sectionDefinition = sections[sectionId];
+                // const questionId = Object.keys(sectionDefinition.schema.properties)[0];
+                // const newProperties = sectionDefinition.schema.properties[questionId];
+                // newProperties.type = 'boolean';
+                // newProperties.onOf = [
+                //     {title: 'Yes', const: true},
+                //     {title: 'No', const: false}
+                // ];
+                // sections[sectionId].schema.properties[questionId] = JSON.parse(
+                //     JSON.stringify(newProperties)
+                // );
+                // // const sectionDefinition = sections[sectionId];
+                // // const questionId = Object.keys(sectionDefinition.schema.properties)[0];
+                // // const currentAnswer = questionnaire.answers[sectionId][questionId];
+                // // const mutatedAnswer = currentAnswer === 'yes';
+                // // questionnaire.answers[sectionId][questionId] = mutatedAnswer;
+
+                const sectionDefinition = sections[sectionId];
```

Comment:
I would have recommended moving these blocks into distinct functions. However, as this code is temporary I think we can let it go?

Reply:
absolutely. It being a matter of time, I am weighing up if it is viable or not.

Reply:
agreed